packages/client/lib/sync/fetcher/blockfetcher.ts

```diff
@@ -24,13 +24,13 @@ export class BlockFetcher extends BlockFetcherBase<Block[], Block> {
   async request(job: Job<JobTask, Block[], Block>): Promise<Block[]> {
     const { task, peer } = job
     const { first, count } = task
-    const headers = await (peer!.eth as EthProtocolMethods).getBlockHeaders({
-      block: first,
-      max: count,
-    })
-    const bodies: BlockBodyBuffer[] = <BlockBodyBuffer[]>(
-      await peer!.eth!.getBlockBodies(headers.map((h) => h.hash()))
-    )
+    const headers = (
+      await (peer!.eth as EthProtocolMethods).getBlockHeaders({
+        block: first,
+        max: count,
+      })
+    )[1]
```

Comment:
if (!headers) {
      this.config.logger.error(`peer ${peer?.id} returned no headers for block ${first}`)
      return
    }